### PR TITLE
Add basic dev setup to README

### PR DESCRIPTION
**Pull request by arothuis**

### CODE REVIEW

**Comment by arothuis** [comment]
@relaxnow @rjkip

File: README.md [hunk 1]
```diff
@@ -49,9 +49,13 @@ Place the SSL certificates in `./Stepup-Deploy/ssl`:
...
 
 ```
-10.10.0.100  mw-dev.stepup.coin.surf.net gw-dev.stepup.coin.surf.net ss-dev.stepup.coin.surf.net ra-dev.stepup.coin.surf.net
+10.10.0.100  mw-dev.stepup.coin.surf.net gw-dev.stepup.coin.surf.net ss-dev.stepup.coin.surf.net ra-dev.stepup.coin.surf.net idp-dev.stepup.coin.surf.net
 10.10.0.101  g2-dev.stepup.coin.surf.net
 ```
 
  * To view the self-service, visit https://ss-dev.stepup.coin.surf.net/app_dev.php/.
- * To view the Graylog2 web interface, vist http://g2-dev.stepup.coin.surf.net:9000 and use username `admin` and password `yourpassword`.
+ * To view the Graylog2 web interface, visit http://g2-dev.stepup.coin.surf.net:9000 and use username `admin` and password `yourpassword`.
```
Comment (rjkip) [inline README.md]: Would it be nice to mention Mailcatcher here as well?

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+      {
+          "sraa": [
+            "9971dbcf01267b11f6107d9cafb43e5b4009a955"
+          ],
+          ...
+      }
+```
+
+To add account `sraa` by its NameID as an SRAA, run the following for Stepup-Middleware, replacing `<YOUR_YUBIKEY_NUMBER>`
```
Comment (rjkip) [inline docs/development.md]: Yubikey number == Yubikey public ID

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+To add account `sraa` by its NameID as an SRAA, run the following for Stepup-Middleware, replacing `<YOUR_YUBIKEY_NUMBER>` 
+with the number relating to your Yubikey (it is usually printed on it):
+
+```
+    app/console middleware:bootstrap:identity-with-yubikey 9971dbcf01267b11f6107d9cafb43e5b4009a955 'Example Inc' 'SRAA' sraa@sraa.example nl_NL <YOUR_YUBIKEY_NUMBER>
+```
+
+To add `raa` and `ra` as RAA and RA respectively, they first have to go through the vetting process.
+Their verification tokens can be registered by `sraa` in RA. The accounts can then also be added as an RAA or RA.
```
Comment (rjkip) [inline docs/development.md]: Beetje ubiquitous language:

verification tokens == second factor tokens
registered == vetted
their identities can also be accredited as RA or RAA

File: README.md (hunk 1 above)
Comment (arothuis) [inline README.md]: Good idea.

File: docs/development.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+Make sure RAA's and RA's are registered conform the LoA required for RA (loa3 by default).
```
Comment (rjkip) [inline docs/development.md]: You're all about commas, I'm all about apostrophes: *RAAs*, *RAs* :yum:

File: README.md [hunk 5]
```diff
@@ -30,7 +30,7 @@ $ git clone git@github.com:SURFnet/Stepup-Middleware.git
 $ git clone git@github.com:SURFnet/Stepup-Gateway.git
 $ git clone git@github.com:SURFnet/Stepup-SelfService.git
 $ git clone git@github.com:SURFnet/Stepup-RA.git
-$ mkdir simplesamlphp # Place a simplesamlphp deployment here with some credentials
+$ mkdir simplesamlphp # Place a simplesamlphp deployment here with some credentials and set the baseurlpath to "/"
```
Comment (relaxnow) [inline README.md]: 👍 this was a real gotcha

File: docs/development.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+      {
+          "sraa": [
+            "9971dbcf01267b11f6107d9cafb43e5b4009a955"
+          ],
+          ...
+      }
+```
+
+To add account `sraa` by its NameID as an SRAA, run the following for Stepup-Middleware, replacing `<YOUR_YUBIKEY_PUBLIC_ID` 
+with the number relating to your Yubikey (it is usually printed on it):
```
Comment (DRvanR) [inline docs/development.md]: also, ensure proper NameID :wink: (in this case it is `9971...`, but when using a different account configured above, you need to use the corresponding NameID)

File: docs/development.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
+# Development setup
```
Comment (DRvanR) [inline docs/development.md]: Might want to mention the FF SAML extension as well as the Chrome SAML extension, really nice when having to debug SAML related things.

File: docs/development.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+Accounts can be added to the `$config` in your git-ignored `simplesamlphp/config/authsources.php` file. 
+For development purposes, it could be useful to add SRAA, RAA and RA accounts as well as users for each
+second factor.
+
+```php
+    ...
+        'sraa:sraa' => array(
+            'commonName'             => 'SRAA',                 // displayName
+            'mail'                    => 'sraa@sraa.example',   // email (also used for EPTI)
+            'schacHomeOrganization'   => 'Example Inc',         // schacHomeOrganisation
```
Comment (relaxnow) [inline docs/development.md]: This is not a valid SHO, SHOs should always be domain names

Comment (relaxnow) [inline docs/development.md]: I'm surprised this works actually.

File: docs/development.md (hunk 6 above)
Comment (arothuis) [inline docs/development.md]: Good call, worth mentioning it.

File: docs/development.md (hunk 7 above)
Comment (arothuis) [inline docs/development.md]: Good call, worth mentioning it. ;)

**Comment by arothuis** [comment]
@rjkip, @relaxnow; merge-worthy?

**Comment by rjkip** [comment]
You go girl :+1:

**Pull request merged**